src/GitVersionCore/BranchConfigurationCalculator.cs

```diff
@@ -18,7 +16,7 @@ public class BranchConfigurationCalculator
             if (matchingBranches.Length == 0)
             {
                 var branchConfig = new BranchConfig();
-                ConfigurationProvider.ApplyBranchDefaults(config, branchConfig);
+                ConfigurationProvider.ApplyBranchDefaults(config, branchConfig, tracksReleaseBranches: true);
```

Comment:
Not sure this is right actually as @DanielRose says. The default shouldn't be to track release branches because there could be feature branches off support branches or any number of other scenarios.